lib/blocs/groups_cubit.dart

```diff
@@ -47,3 +51,52 @@ class CommitteesCubit extends GroupsCubit {
 class SocietiesCubit extends GroupsCubit {
   SocietiesCubit(ApiRepository api) : super(api, MemberGroupType.society);
 }
+
+class AllGroupsCubit extends GroupsCubit {
+  /// The last used search query. Can be set through `this.search(query)`.
+  String? _searchQuery;
+
+  /// A timer used to debounce calls to `this.load()` from `this.search()`.
+  Timer? _searchDebounceTimer;
+
+  AllGroupsCubit(ApiRepository api) : super(api, null);
+
+  @override
+  Future<void> load() async {
+    // print("AAA");
+    emit(const LoadingState());
+    try {
+      final query = _searchQuery;
+
+      if (query != _searchQuery) return;
```

Comment:
This check if query changed should be called after getting the response from the api